### PR TITLE
Removed subgroups as child elements from getDataWithGroups

### PR DESCRIPTION
**Pull request by metodej**
- subgroups are no longer child elements of structure returned by getDataWithGroups, instead there is empty first child element in group part of the structure so current services does not have to be changed

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java [hunk 1]
```diff
@@ -549,10 +549,6 @@
 	 |              |       |        +-------Attrs
 	 |              |       |        +-------ChildNodes
 	 |              |       |                   +-------()
-	 |              |       |                   |        +---ChildNodes
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java:552]: This part of JavaDoc is needed to be removed also a few lines above.

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java [hunk 2]
```diff
@@ -519,7 +519,7 @@
 	 * @return attributes in special structure. Facility is in the root, facility children are resources.
 	 *         Resource first child is abstract structure which children are groups.
 	 *         Resource  second chi is abstract structure which children are members.
-	 *         Group first child is abstract structure which children are groups.
+	 *         Group first child is empty structure.
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java:522]: Please add also some short info about why is the empty structure needed.

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/ServicesManagerBl.java [hunk 3]
```diff
@@ -452,10 +452,8 @@
 	 * @return attributes in special structure. Facility is in the root, facility children are resources.
 	 *         Resource first child is abstract structure which children are groups.
 	 *         Resource  second chi is abstract structure which children are members.
-	 *         Group first child is abstract structure which children are groups.
+	 *         Group first child is empty structure.
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/ServicesManagerBl.java:455]: Same as above

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

**Review by stavamichal — APPROVED** [review]

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java:552]: Ok, thanks. I have also updated the javadoc of rpc.

**Review by balcirakpeter — APPROVED** [review]

**Pull request merged**